crates/bitwarden/src/tool/generators/password.rs

```diff
@@ -44,6 +45,51 @@ pub(super) fn password(_input: PasswordGeneratorRequest) -> Result<String> {
     Ok("pa11w0rd".to_string())
 }
 
-pub(super) fn passphrase(_input: PassphraseGeneratorRequest) -> Result<String> {
-    Ok("correct-horse-battery-staple".to_string())
+const DEFAULT_PASSPHRASE_NUM_WORDS: u8 = 3;
+const DEFAULT_PASSPHRASE_SEPARATOR: char = ' ';
+
+pub(super) fn passphrase(input: PassphraseGeneratorRequest) -> Result<String> {
+    let words = input.num_words.unwrap_or(DEFAULT_PASSPHRASE_NUM_WORDS);
+    let separator = input
+        .word_separator
+        .and_then(|s| s.chars().next())
+        .unwrap_or(DEFAULT_PASSPHRASE_SEPARATOR);
+
+    let capitalize = input.capitalize.unwrap_or(false);
+    let include_number = input.include_number.unwrap_or(false);
+
+    fn capitalize_first_letter(s: &str) -> String {
+        let mut c = s.chars();
+        match c.next() {
+            None => String::new(),
+            Some(f) => f.to_uppercase().collect::<String>() + c.as_str(),
+        }
+    }
+
+    let mut rand = rand::thread_rng();
+
+    let insert_number_idx = include_number.then(|| rand.gen_range(0..words));
+    let mut passphrase = String::new();
+
+    for idx in 0..words {
+        let word = EFF_LONG_WORD_LIST
+            .choose(&mut rand)
+            .expect("slice is not empty");
+
+        if capitalize {
+            passphrase.push_str(&capitalize_first_letter(word));
+        } else {
+            passphrase.push_str(word);
+        }
+
+        if insert_number_idx == Some(idx) {
+            passphrase.push_str(&rand.gen_range(0..=9).to_string());
+        }
+
+        if idx != words - 1 {
+            passphrase.push(separator)
+        }
+    }
```

Comment:
**suggestion:** I think we can break this up into a few more easily understood functions that can be more easily tested.

```rust
/// Generate random words from the dictionary
fn gen_words(mut rng: impl RngCore, num_words: u8) -> Vec<String> {
    (0..num_words)
        .map(|_| {
            EFF_LONG_WORD_LIST
                .choose(&mut rng)
                .expect("slice is not empty")
                .to_string()
        })
        .collect()
}
```

This removes the need for a loop, adding a number means updating the index of the affected string, and the word string is a join using the separator as the joiner.